app/controllers/twitter_users_controller.rb

```diff
@@ -0,0 +1,79 @@
+class TwitterUsersController < ApplicationController
+  before_action :current_user
+
+  def subscribe
+    @screen_name = params[:screen_name]
+    if twitter_user_protected?
+      flash[:error] = "That user is protected; you cannot subscribe to their tweets."
+    else
+      if !twitter_user_exists?
+        # if twitter_user doesn't exist, create twitter_user and their tweets
+        create_twitter_user(@screen_name)
+        create_tweets(@twitter_user)
+      end
+      @twitter_user = TwitterUser.find_by(screen_name: @screen_name)
+      if @current_user.twitter_users.include?(@twitter_user)
+        #if current_user is already subscribed to this twitter_user, show error
+        flash[:error] = "You are already subscribed to this user."
+      else
+        # subscribe current_user to twitter_user
+        @current_user.twitter_users << @twitter_user
+      end
+    end
+    redirect_to :root
+  end
+
+  def unsubscribe
+    @screen_name = params[:screen_name]
+    @twitter_user = TwitterUser.find_by(screen_name: @screen_name)
+    @current_user.twitter_users.delete(@twitter_user)
+    redirect_to :subscriptions
+  end
+
+  private
+
+  def twitter_user_exists?
+    if TwitterUser.find_by(screen_name: @screen_name)
+      return true
+    else
+      return false
+    end
```

Comment:
This is a great opportunity to use the ternary operator to reduce this method to one line, as such: 
`return TwitterUser.find_by(screen_name: @screen_name) ? true : false`

This is saying, if the user is nil, return false, otherwise return true (more on the ternary operator: http://stackoverflow.com/questions/4252936/how-do-i-use-the-conditional-operator-in-ruby)